src/autoscaler/eventgenerator/cmd/eventgenerator/eventgenerator_suite_test.go

```diff
@@ -0,0 +1,205 @@
+package main_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"autoscaler/db"
+	"autoscaler/eventgenerator/config"
+	"autoscaler/models"
+	"database/sql"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	"github.com/onsi/gomega/ghttp"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"testing"
+	"time"
+)
+
+var (
+	egPath          string
+	testAppId       string = "testAppId"
+	metricType      string = "MemoryUsage"
+	timestamp       int64  = time.Now().UnixNano()
+	regPath                = regexp.MustCompile(`^/v1/apps/.*/scale$`)
+	configFile      *os.File
+	conf            *config.Config
+	metricCollector *ghttp.Server
+	scalingEngine   *ghttp.Server
+	metrics         []*models.Metric = []*models.Metric{
+		&models.Metric{
+			Name:      metricType,
+			Unit:      "bytes",
+			AppId:     testAppId,
+			TimeStamp: timestamp,
+			Instances: []models.InstanceMetric{models.InstanceMetric{
+				Timestamp: timestamp,
+				Index:     0,
+				Value:     "500",
+			}, models.InstanceMetric{
+				Timestamp: timestamp,
+				Index:     1,
+				Value:     "600",
+			}},
+		},
+		&models.Metric{
+			Name:      metricType,
+			Unit:      "bytes",
+			AppId:     testAppId,
+			TimeStamp: timestamp,
+			Instances: []models.InstanceMetric{models.InstanceMetric{
+				Timestamp: timestamp,
+				Index:     0,
+				Value:     "700",
+			}, models.InstanceMetric{
+				Timestamp: timestamp,
+				Index:     1,
+				Value:     "800",
+			}},
+		},
+	}
+)
+
+func TestEventgenerator(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Eventgenerator Suite")
+}
+
+var _ = SynchronizedBeforeSuite(func() []byte {
+	eg, err := gexec.Build("autoscaler/eventgenerator/cmd/eventgenerator", "-race")
+	Expect(err).NotTo(HaveOccurred())
+	return []byte(eg)
+}, func(pathByte []byte) {
+	egPath = string(pathByte)
```

Comment:
Can you break this down into separate methods so it's obvious what each grouping does, e.g., initHttpEndpoints(), initDB(), etc...